### DISCUSSION TITLE
Use of tracy with hpx

### DISCUSSION THREAD

**Discussion by alexanderscheible** [STEllAR-GROUP/hpx#6571] [Q&A]
Hello together, 

does anyone have experience with using the profiler [tracy](https://github.com/wolfpld/tracy) together with hpx?

I am working with a code base that currently uses a very simple thread pool (https://github.com/bshoshany/thread-pool), but I am considering switching to hpx because we are reaching the thread pools limits. Switching the functionality to hpx has been actually pretty easy (mainly using the parallel variants of hpx::ranges::*), but I have not found a way to get tracy working although I suspect this should be possible as tracy has fiber support.

Thanks a lot,
Alex

**Comment by hkaiser** [comment]
I don't know anything about tracy myself, but I'll try to find the time to have a look. We have integrated other tools before, so I don't think this is a major issue.

**Comment by hkaiser** [comment]
From what I can see, we could integrate Tracy with HPX to: 

- name kernel threads such that tracy can identify those
- mark HPX thread execution (start, suspension, resumption, termination), annotate zones using the HPX thread description (using Tracy's fiber API)
- HPX lock handling (acquire, release), annotate locks
- monitoring memory usage (HPX's internal memory pools, etc.)
- ... not sure if there is more

Would you be interested in working on this? We'd be more than happy to guide you and to help, if needed.

**Comment by Pansysk75** [comment]
I was actually looking for something like tracy last week, so I'm interested in this. Thanks Alex for bringing this into my attention :)

**Comment by hkaiser** [comment]
See here for a first version of integrating HPX with Tracy: https://github.com/STEllAR-GROUP/hpx/pull/6858
